iothub/device/devdoc/Convention-based operations.md

```diff
@@ -0,0 +1,213 @@
+    public void AddOrUpdate(string propertyName, object propertyValue, int statusCode, long version, string description = null, string componentName = null);
+    public void AddOrUpdate(string propertyName, object propertyValue, string componentName);
+    public bool Contains(string componentName, string propertyName);
+    public virtual bool TryGetValue<T>(string componentName, string propertyName, out T propertyValue);
+}
+
+public interface IWritablePropertyResponse {
+    int AckCode { get; set; }
+    string AckDescription { get; set; }
+    long AckVersion { get; set; }
+    object Value { get; set; }
+}
+
+public sealed class NewtonsoftJsonWritablePropertyResponse : IWritablePropertyResponse {
+    public NewtonsoftJsonWritablePropertyResponse(object propertyValue, int ackCode, long ackVersion, string ackDescription = null);
+    public int AckCode { get; set; }
+    public string AckDescription { get; set; }
+    public long AckVersion { get; set; }
+    public object Value { get; set; }
+}
+
+public class ClientPropertiesUpdateResponse {
+    public ClientPropertiesUpdateResponse();
+    public string RequestId { get; internal set; }
+    public long Version { get; internal set; }
+}
+```
+
+### Telemetry
+
+```csharp
+/// <summary>
+/// Send telemetry using the specified message.
+/// </summary>
+/// <remarks>
+/// Use the <see cref="TelemetryMessage(string, TelemetryCollection)"/> constructor to pass in the optional
+/// <see cref="TelemetryCollection"/> that specifies your payload and serialization and encoding rules.
+/// </remarks>
+/// <param name="telemetryMessage">The telemetry message.</param>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+/// <returns></returns>
+public Task SendTelemetryAsync(TelemetryMessage telemetryMessage, CancellationToken cancellationToken = default);
+```
+#### All related types
+
+```csharp
+public class TelemetryCollection : PayloadCollection {
+    public TelemetryCollection();
+    public override void Add(string telemetryName, object telemetryValue);
+    public override void AddOrUpdate(string telemetryName, object telemetryValue);
+}
+
+public class TelemetryMessage : Message {
+    public TelemetryMessage(string componentName = null);
+    public new string ContentEncoding { get; internal set; }
+    public new string ContentType { get; internal set; }
+    public TelemetryCollection Telemetry { get; set; }
+    public override Stream GetBodyStream();
+}
+```
+
+### Commands
+
+```csharp
+/// <summary>
+/// Set the global command callback handler.
+/// </summary>
+/// <param name="callback">A method implementation that will handle the incoming command.</param>
+/// <param name="userContext">Generic parameter to be interpreted by the client code.</param>
+/// <param name="payloadConvention">A convention handler that defines the content encoding and serializer to use for commands.</param>
```

Comment:
Is this param not part of the API? - payloadConvention

Reply:
it is not, I'll update this